Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -51,10 +94,58 @@ public static int[] RemoveFirst(int[] array)
         /// <returns>A new array with element removed at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveAt(int[] array, int index)
         {
-            // ToDo: implement.
+            var newArray = ProcessArray(array, index);
+            return newArray;
+        }
+
+        private static int[] ProcessArray(int[] array, int index)
+        {
+            var isValid = IsValidCase(array, index);
+
+            if (!isValid)
+            {
+                return array;
+            }
+
+            array = Remove(array, index);
             return array;
         }
 
+        private static bool IsValidCase(int[] array, int index)
+        {
+            if (array == null) 
+            {
+                return false;
+            }
+
+            if (index < 0 || index == array.Length || array.Length == 0 || index > array.Length)
```

Comment:
I would maybe put array.Length == 0 behind the upper check (array == null). This separates array checks from the index checks. :)

Reply:
done